### ISSUE TITLE
Is there a way to emit parser failure from my parsers?

### ISSUE THREAD

**Issue by loxs** [microsoft/ts-parsec#15]
I am trying to achieve this, but I can't figure out what's the proper solution:

```ts
const dateParser = apply(tok(TokenKind.ISODate), (t) => {
  const parsedDate = parseISO(t.text);
  if (isValid(parsedDate)) {
    return parsedDate;
  }
  return; //what?;
});
```

What is the proper idiomatic solution to this case?

**Comment by vczh** [comment]
@loxs Sorry to be so late to see this issue, since this project is not actively updated.

This is in my plan. You are not able to do it elegantly for now.

**Comment by ZihanChen-MSFT** [comment]
`err` and `errd` are available in 0.3.2, please checkout the document.
